subsys/bootloader/cmake/sign.cmake

```diff
@@ -117,45 +55,124 @@ if (CONFIG_SB_PRIVATE_KEY_PROVIDED)
   )
 endif()
 
-add_custom_command(
-  OUTPUT
-  ${signed_hex}
-  COMMAND
-  ${PYTHON_EXECUTABLE}
-  ${NRF_BOOTLOADER_SCRIPTS}/validation_data.py
-  --input ${to_sign}
-  --output ${signed_hex}
-  --offset ${CONFIG_FW_VALIDATION_METADATA_OFFSET}
-  --signature ${SIGNATURE_FILE}
-  --public-key ${SIGNATURE_PUBLIC_KEY_FILE}
-  --magic-value "${VALIDATION_INFO_MAGIC}"
-  DEPENDS
-  ${SIGN_KEY_FILE_DEPENDS}
-  WORKING_DIRECTORY
-  ${PROJECT_BINARY_DIR}
-  COMMENT
-  "Creating validation for ${KERNEL_HEX_NAME}, storing to ${SIGNED_KERNEL_HEX_NAME}"
-  USES_TERMINAL
-  )
+include(${CMAKE_CURRENT_LIST_DIR}/../cmake/fw_info_magic.cmake)
 
-add_custom_target(
-  signed_kernel_hex_target
-  DEPENDS
-  ${signed_hex}
-  signature_file_target
-  signature_public_key_file_target
-  )
+set(slots s0)
+
+if (CONFIG_MCUBOOT_BUILD_S1_VARIANT)
+  list(APPEND slots s1)
+endif ()
+print(slots)
+
+foreach (slot ${slots})
+  set(signed_hex ${PROJECT_BINARY_DIR}/signed_by_b0_${slot}.hex)
+  set(sign_depends ${slot}_hex)
+  set(to_sign ${PROJECT_BINARY_DIR}/${slot}.hex)
```

Comment:
How are these hex files created in the first place?

Reply:
s0 is automatically created by partition manager. See https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/186 for details on how s1 is built.